tests/test_frozenlist.py

```diff
@@ -273,6 +274,20 @@ def test_deepcopy_frozen(self) -> None:
         with pytest.raises(RuntimeError):
             copied.append(4)
 
+    def test_deepcopy_frozen_circular(self) -> None:
+        orig = self.FrozenList([1, 2])
+        orig.append(orig)  # Create circular reference
+        orig.freeze()
+        copied = deepcopy(orig)
+        assert copied[0] == 1
+        assert copied[1] == 2
+        assert len(copied[2]) == 3
+        assert copied[2][0] == 1
+        assert copied[2][1] == 2
+        assert len(copied[2][2]) == 3
+        # ... and so on. Testing equality when a structure includes itself is tough.
```

Comment:
Wouldn't `assert copied[2] is copied` accurately test this?

Reply:
Yes. I knew I was confusing myself. We deepcopy once, then its in the memo so it reuses at that and therefore references itself. Thanks!